### DISCUSSION TITLE
Non-SMTP email configuration

### DISCUSSION THREAD

**Discussion by ddeepwell** [apache/airflow#32834] [Q&A] [answered]
I know how to configure airflow to send emails with SMTP, but I now have an email address which cannot use the SMTP AUTH protocol. Is there an OAuth2 authentication method or some other non-legacy method? Microsoft has started to [deprecate basic authentication](https://learn.microsoft.com/en-us/exchange/clients-and-mobile-in-exchange-online/deprecation-of-basic-authentication-exchange-online?source=recommendations).

How do I configure the email settings in Apache Airflow so that emails (on task retries, failure) are sent with an Office 365 account which cannot use the SMTP AUTH protocol?

**Comment by potiuk** [comment, marked as answer]
I don't think this is possible. But any conttibutions are welcome. You can write your own notification service - which is close enough https://airflow.apache.org/docs/apache-airflow/stable/howto/notifications.html - but if you think having different methods of authentication for sending emails is a good idea, PRs are always most welcome.

Generally speaking you can for exmple set-up your own local mailserver that will accept connections only from Airflow and use basic authentication there, and let your serve communicate with other servces using different authentication mechanisms, so you are not entirely blocked, even if it is a bit of a hassle.

But if you think contributing different mechanims is a good, idea, then I think it's a good one to follow.
